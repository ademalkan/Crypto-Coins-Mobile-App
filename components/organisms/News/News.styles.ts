import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    padding: 16,
    backgroundColor: '#F8F8F8',
    width: '100%',
  },
  newsCard: {
    backgroundColor: 'white',
    borderRadius: 8,
    padding: 16,
    marginBottom: 16,
  },

  newsTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 8,
  },
  newsDescription: {
    fontSize: 16,
    marginBottom: 8,
  },
  newsSource: {
    fontSize: 14,
    color: 'gray',
  },
})

