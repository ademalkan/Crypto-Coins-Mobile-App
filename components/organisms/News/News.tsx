import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { CryptoNewsI } from '../../../interfaces/CryptoNewsInterface';
import ImageAtom from '../../atoms/ImageAtom/ImageAtom';
import styles from './News.styles';

interface NewsProps {
  item: CryptoNewsI;
}

const News: React.FC<NewsProps> = ({ item }) => {
  const navigation = useNavigation();

  const handlePress = () => {
    navigation.navigate('NewsDetail', { news_title: item.title });
  };

  return (
    <TouchableOpacity onPress={handlePress}>
      <View style={styles.newsCard}>
        <ImageAtom image={item.thumb_2x} height={200} />
        <Text style={styles.newsTitle}>{item.title}</Text>
        <Text style={styles.newsDescription}>
          {item.description.length > 60
            ? `${item.description.slice(0, 60)}...`
            : item.description}
        </Text>
        <Text style={styles.newsSource}>{item.author}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default News;
