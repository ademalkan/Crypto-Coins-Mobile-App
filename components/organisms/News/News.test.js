import React from "react";
import renderer from "react-test-renderer";
import News from "./News";

jest.mock("../../../context/CryptoNewsContext", () => ({
  CryptoNewsContext: {
    cryptoNews: [
      {
        id: "1",
        title: "Bitcoin News",
        description: "Lorem ipsum dolor sit amet.",
        author: "John Doe",
        thumb_2x: "news.png",
        url: "https://example.com/news/1",
      },
      {
        id: "2",
        title: "Ethereum News",
        description: "Lorem ipsum dolor sit amet.",
        author: "Jane Smith",
        thumb_2x: "news.png",
        url: "https://example.com/news/2",
      },
    ],
  },
}));

describe("News", () => {
  it("should render the news component correctly", () => {
    const component = renderer.create(<News />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
