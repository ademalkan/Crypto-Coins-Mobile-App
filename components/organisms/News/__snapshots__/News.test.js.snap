// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News should render the news component correctly 1`] = `
<RCTScrollView
  data={[]}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View />
</RCTScrollView>
`;
